newhelm/single_turn_prompt_response.py

```diff
@@ -30,14 +33,14 @@ def context(self):
     context_internal: _Context = None
     """Internal variable for the serialization friendly version of context"""
 
-    def __init__(self, *, prompt, context=None, context_internal=None):
+    def __init__(self, *, prompt, source_id=None, context=None, context_internal=None):
         if context_internal is not None:
             internal = context_internal
         elif isinstance(context, BaseModel):
             internal = TypedData.from_instance(context)
         else:
             internal = context
-        super().__init__(prompt=prompt, context_internal=internal)
+        super().__init__(prompt=prompt, souce_id=source_id, context_internal=internal)
```

Comment:
I think you meant `source_id=source_id`

Reply:
Great catch! I fixed this typo and added better test coverage.